merlin/dag/executors.py

```diff
@@ -379,6 +437,24 @@ def fit(self, dataset: Dataset, graph, strict=False):
         for computed_stats, node in zip(results, nodes):
             node.op.fit_finalize(computed_stats)
 
+    def _transform_impl(self, dataset: Dataset, graph: Graph, capture_dtypes=False):
```

Comment:
This came from NVT, where the name `transform_impl` made more sense. Debating what a better name for this would be (or if it should just be inlined.)

Reply:
Looks like we may not need most of this method since the `self.transform` now always returns a Dataset and the graph construction is already applied in the `fit` method were it's called. So inlining might only require the line that calls `self.transform`?